### PR TITLE
[KYUUBI #5417] should not check in-subquery in permanent view

### LINKED ISSUE #5417: [Bug] Authz will still check source table when persist view contains a subquery
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

```
          sql(s"CREATE TABLE $db.$v1 AS SELECT * FROM $view")
          sql(s"CREATE TABLE $db.$v2 AS SELECT * FROM $view")
          sql(
            s"""
               |CREATE VIEW $db.$my_view
               |AS
               |WITH temp AS (
               |    SELECT max(grade) max_grade
               |    FROM $db.$v2)
               |SELECT id as new_id FROM $db.$v1
               |WHERE grade = (SELECT max_grade FROM temp)
               |""".stripMargin)
          sql(
            s"SELECT * FROM $db.$my_view".stripMargin).explain(true)
```

For such a case,  if the view have a subquery, spark will first execute subquery  then  it may still check the internal table of persist view.

```
== Analyzed Logical Plan ==
new_id: int
Project [new_id#54]
+- SubqueryAlias spark_catalog.test_default.my_view
   +- PermanentViewMarker `test_default`.`my_view`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
      +- View (`test_default`.`my_view`, [new_id#54])
         +- Project [cast(new_id#51 as int) AS new_id#54]
            +- WithCTE
               :- CTERelationDef 1
               :  +- SubqueryAlias temp
               :     +- Aggregate [max(grade#57) AS max_grade#53]
               :        +- SubqueryAlias spark_catalog.test_default.v2
               :           +- HiveTableRelation [`test_default`.`v2`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [id#55, name#56, grade#57, sex#58], Partition Cols: []]
               +- Project [id#59 AS new_id#51]
                  +- Filter (grade#61 = scalar-subquery#52 [])
                     :  +- Project [max_grade#53]
                     :     +- SubqueryAlias temp
                     :        +- CTERelationRef 1, true, [max_grade#53]
                     +- SubqueryAlias spark_catalog.test_default.v1
                        +- HiveTableRelation [`test_default`.`v1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [id#59, name#60, grade#61, sex#62], Partition Cols: []]

```
### Affects Version(s)

master, 1.8.0

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
Fix #5417 
If there is is a view with in-subquery, authz will still request this in-subquery's interval privilege, it's not we want.
For view
```
CREATE VIEW db.view1
AS
WITH temp AS (
    SELECT max(scope) max_scope
    FROM db1.table1)
SELECT id as new_id FROM db1.table2
WHERE scope in (SELECT max_scope FROM temp)
```

When we query the view 
```
SEELCT * FROM db.view1
```

Before this pr, since spark will first execute subquery, it will first request `[default/table1/scope]` then request `[default/view1/new_id]`

after this pr, it only request  `[default/view1/new_id]`

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
no

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala [hunk 1]
```diff
@@ -43,6 +43,8 @@ class RuleApplyPermanentViewMarker extends Rule[LogicalPlan] {
             //  as the main query part also secures it. But for performance consideration,
             //  we also pre-check it in subqueries and fail fast with negative privileges.
             scalarSubquery.copy(plan = PermanentViewMarker(scalarSubquery.plan, null))
+          case listQuery: ListQuery =>
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala:46]: use SubqueryExpression for both ScalarSubquery and ListQuery?

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala:46]: > SubqueryExpression

Then can't use copy method.

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala:46]: You can use withNewPlan instead

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyPermanentViewMarker.scala]: > You can use withNewPlan instead

Cool ! changed.

**Review by yaooqinn — APPROVED** [review]
LGTM, pending CI.

**Comment by yaooqinn** [comment]
Thanks @AngersZhuuuu, merged to master

**Comment by bowenliang123** [comment]
A late LGTM. Thanks for fixing this. Is that possible to backport to 1.8?

**Pull request closed without merging**